index.html

```diff
@@ -101,212 +225,1327 @@ <h3>
+                <ol>
+                  <li>
+                    Invoke the operating system to shutdown the socket for writing.
+                  <li>
+                    [=Queue a global task=] on the [=relevant global object=] of
+                    [=this=] using the [=TCP socket task source=] to run the
+                    following steps:
+                    <ol>
+                      <li>
+                        Invoke the steps to [=handle closing the TCP writable stream=].
+                      <li>
+                        If |signal| is [=AbortSignal/aborted=], [=reject=] |promise|
+                        with |signal|'s [=AbortSignal/abort reason=].
+                      <li>
+                        [=Resolve=] |promise| with `undefined`.
+                    </ol>
+                </ol>
+              <li>
+                Return |promise|.
+            </ol>
+          <li>
+            [=WritableStream/Set up=] |stream| with
+            <a href="https://streams.spec.whatwg.org/#writablestream-set-up-writealgorithm">writeAlgorithm</a>
+            set to |writeAlgorithm|,
+            <a href="https://streams.spec.whatwg.org/#writablestream-set-up-abortalgorithm">abortAlgorithm</a>
+            set to |abortAlgorithm|,
+            <a href="https://streams.spec.whatwg.org/#writablestream-set-up-closealgorithm">closeAlgorithm</a>
+            set to |closeAlgorithm|,
+            <a href="https://streams.spec.whatwg.org/#writablestream-set-up-highwatermark">highWaterMark</a>
+            set to an implementation-defined value.
+          <li>
+            [=AbortSignal/Add=] the following abort steps to |signal|:
+            <ol>
+              <li>
+                Cause any invocation of the operating system to write to the
+                socket to return as soon as possible no matter how much data has
+                been written.
+            </ol>
+          <li>Set [=this=].{{TCPSocket/[[writable]]}} to |stream|.
+        </ol>
+
+        To <dfn>handle closing the TCP writable stream</dfn> perform the following steps:
+
+        <ol>
+          <li>
+            If [=this=].{{TCPSocket/[[readable]]}} is active, ignore the following steps.
+          <li>
+            Run the following steps [=in parallel=].
+            <ol>
+              <li>
+                Invoke the operating system to close the socket.
+              <li>
+                [=Queue a global task=] on the [=relevant global object=] of
+                [=this=] using the [=TCP socket task source=] to run the
+                following steps:
+                <ul>
+                  <li>
+                    If [=this=].{{TCPSocket/[[writable]]}} is errored,
+                    [=reject=] [=this=].{{TCPSocket/[[closedPromise]]}}
+                    with [=this=].{{TCPSocket/[[writable]]}}.`[[storedPromise]]`.
+                  <li>
+                    Otherwise, if [=this=].{{TCPSocket/[[readable]]}} is errored,
```

Comment:
Check this first, since we're in the case where the readable stream closed first its error should have priority.